StoryMap/StoryMap/Screens/Map/Coordinator/MapCoordinator.swift

```diff
@@ -7,62 +7,93 @@
 
 import Foundation
 import UIKit
+import Combine
 
 class MapCoordinator: CoordinatorType {
-    var presenter = UINavigationController()
+	var presenter: UINavigationController
     
-    var addStoryCoordinator: AddStoryCoordinator?
     var storyDetailCoordinator: StoryDetailCoordinator?
     var storyListCoordinator: StoryListCoordinator?
-    
-    var onDidStop: (() -> Void)?
 	
-	private var onDeleteStory: ((Story) -> Void)?
+	private var deleteStorySubject: PassthroughSubject<Story, Never>?
+	
+	private let photoManager = PhotoInputManager()
+	private var subscribers = Set<AnyCancellable>()
+	
+	private var photoManagerSubscriber: AnyCancellable?
+	
+	init(presenter: UINavigationController) {
+		self.presenter = presenter
+	}
     
-    func start(_ presentFrom: UIViewController?) {
+    func start() {
         let viewModel = MapViewModel()
         let viewController = MapViewController(
             viewModel: viewModel,
             locationManager: LocationManager()
         )
         viewController.modalPresentationStyle = .fullScreen
-        viewModel.onAddStory = { [weak self] location in
-            self?.showAddStory(with: location)
-        }
-        viewModel.onOpenStory = { [weak self] story in
-            self?.showStoryDetail(with: story)
-        }
-        viewModel.onOpenStoryList = { [weak self] in
-            self?.showStoryList()
-        }
-		onDeleteStory = viewModel.storyDeleted
+        viewModel.addStorySubject
+			.sink { [weak self] location in
+				self?.showAddStory(with: location)
+			}
+			.store(in: &subscribers)
+        viewModel.openStorySubject
+			.sink { [weak self] story in
+				self?.showStoryDetail(with: story)
+			}
+			.store(in: &subscribers)
+        viewModel.openStoryListSubject
+			.sink { [weak self] in
+				self?.showStoryList()
+			}
+			.store(in: &subscribers)
+		
+		deleteStorySubject = viewModel.storyDeletedSubject
```

Comment:
I wonder if we have a memory leak here, if we keep the instance.
Did you check?